azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.autoconfigure.aad;
+
+import com.microsoft.aad.adal4j.ClientCredential;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserRequest;
+import org.springframework.security.oauth2.client.oidc.userinfo.OidcUserService;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.user.DefaultOidcUser;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+
+import javax.naming.ServiceUnavailableException;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+
+@Configuration
+@ConditionalOnWebApplication
+@ConditionalOnProperty(prefix = "azure.activedirectory", value = {"tenant-id"})
+@EnableConfigurationProperties({AADAuthenticationProperties.class, ServiceEndpointsProperties.class})
+public class AADOAuth2AutoConfiguration {
+    private AADAuthenticationProperties aadAuthProps;
+    private ServiceEndpointsProperties serviceEndpointsProps;
+
+    public AADOAuth2AutoConfiguration(AADAuthenticationProperties aadAuthProperties,
+                                      ServiceEndpointsProperties serviceEndpointsProps) {
+        this.aadAuthProps = aadAuthProperties;
+        this.serviceEndpointsProps = serviceEndpointsProps;
+    }
+
+    @Bean
+    @ConditionalOnProperty(prefix = "azure.activedirectory", value = {"ActiveDirectoryGroups"})
+    public OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService() {
+        return this.delegationOidcUserService();
+    }
+
+    private OAuth2UserService<OidcUserRequest, OidcUser> delegationOidcUserService() {
+        final OidcUserService delegate = new OidcUserService();
+
+        return (userRequest) -> {
```

Comment:
I'd move that to its own class.

Reply:
OK, thank you, will move to separate class to make it more clear.